### PR TITLE
feat(server): redzone

### PR DESCRIPTION
**Pull request by lmssiehdev**

### CODE REVIEW

File: server/src/objects/gas.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+    countdownStart = 0;
+
+    /** Gas Timer */
+    gasT = 0;
+
+    dirty = true;
+    timeDirty = true;
+
+    get data(): GasData {
```
Comment (leia-uwu) [inline server/src/objects/gas.ts]: why this getter? just set the packet gasData to the class instance

File: server/src/objects/gas.ts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        return math.distanceBetween(pos, this.currentPos) >= this.currentRad;
+    }
+}
+
+const GasStages = [
+    {
+        mode: GasMode.Inactive,
+        duration: 0,
+        radOld: 534.6,
+        radNew: 534.6,
```
Comment (leia-uwu) [inline server/src/objects/gas.ts]: gas radius should be based on map size (a value between 0 and 1) not hardcoded values

File: shared/utils/math.ts [hunk 3]
```diff
@@ -10,6 +10,15 @@ export const math = {
         return a < max ? a > min ? a : min : max;
     },
 
+    /**
+     * Generate a random floating-point value.
+     * @param min The minimum value that can be generated.
+     * @param max The maximum value that can be generated.
+    */
+    randomFloat(min: number, max: number): number {
```
Comment (leia-uwu) [inline shared/utils/math.ts]: we already have a random float function on util.ts
https://github.com/leia-uwu/resurviv/blob/master/shared/utils/util.ts#L36

**Review by leia-uwu — CHANGES_REQUESTED** [review]

File: server/src/objects/gas.ts [hunk 4]
```diff
@@ -138,119 +127,119 @@ const GasStages = [
     {
         mode: GasMode.Inactive,
         duration: 0,
-        radOld: 534.6,
-        radNew: 534.6,
+        radOld: 0.7,
```
Comment (lmssiehdev) [inline server/src/objects/gas.ts:130]: 0.64 ( value in the wiki ) was showing the red zone even when it's inactive

File: shared/utils/math.ts [hunk 5]
```diff
@@ -231,6 +232,31 @@ export const math = {
         return inside;
     },
 
+    /**
+     * Calculate the distance between two points.
+     * @param v1 The first point.
+     * @param v2 The second point.
+     */
+    distanceBetween(v1: Vec2, v2: Vec2): number {
```
Comment (leia-uwu) [inline shared/utils/math.ts]: theres `v2.distance`

**Pull request merged**